nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-api/src/main/java/org/apache/nifi/web/api/TemplateResource.java

```diff
@@ -168,7 +172,13 @@ public Response exportTemplate(
         }
 
         // generate the response
-        return generateOkResponse(template).header("Content-Disposition", String.format("attachment; filename=\"%s.xml\"", attachmentName)).build();
+        /*
+         * Here instead of relying on default JAXB marshalling we are simply
+         * serializing template to String (formatted, indented etc) and sending
+         * it as part of the response.
+         */
+        String serializedTemplate = new String(TemplateSerializer.serialize(template), StandardCharsets.UTF_8);
```

Comment:
This is fine here but we could have stream the template instead of converting to a String in memory using an approach like [1]

[1] https://github.com/apache/nifi/blob/5cd5a4ce788bec39e0dc5c7db93356610a9fa84e/nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-api/src/main/java/org/apache/nifi/web/api/ProvenanceEventResource.java#L146